blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/EnumValidator.java

```diff
@@ -0,0 +1,167 @@
+import com.netflix.astyanax.model.ColumnList;
+import com.rackspacecloud.blueflood.concurrent.ThreadPoolBuilder;
+import com.rackspacecloud.blueflood.io.AstyanaxReader;
+import com.rackspacecloud.blueflood.io.AstyanaxWriter;
+import com.rackspacecloud.blueflood.io.DiscoveryIO;
+import com.rackspacecloud.blueflood.io.SearchResult;
+import com.rackspacecloud.blueflood.types.BluefloodEnumRollup;
+import com.rackspacecloud.blueflood.types.IMetric;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.PreaggregatedMetric;
+import com.rackspacecloud.blueflood.utils.ModuleLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+/** This class handles collecting enum values for specific metrics, via their locators, and checking whether
+    a uniqueness count of the enum values have reached a certain threshold.  If it has, the class mark the metric
+    as bad by inserting its locator into the metrics_excess_enums table.  If it hasn't reached the threshold, then the class
+    will create/update the elastic search "enums" index for the metric, which will contain a list of all unique
+    enum values that has been ingested for it.
+**/
+public class EnumValidator implements Runnable {
+
+    private static final Logger log = LoggerFactory.getLogger(EnumValidator.class);
+    private static final Configuration config = Configuration.getInstance();
+    private static final DiscoveryIO discoveryIO = (DiscoveryIO) ModuleLoader.getInstance(DiscoveryIO.class, CoreConfig.ENUMS_DISCOVERY_MODULES);
+    private static final int ENUM_UNIQUE_VALUES_THRESHOLD = config.getIntegerProperty(CoreConfig.ENUM_UNIQUE_VALUES_THRESHOLD);
+    private static final int ENUM_VALIDATOR_THREADS = Configuration.getInstance().getIntegerProperty(CoreConfig.ENUM_VALIDATOR_THREADS);
+
+    HashSet<Locator> locators;
+
+    public EnumValidator(Set<Locator> locators) {
+        this.locators = (HashSet<Locator>) locators;
+    }
+
+    @Override
+    public void run() {
+
+        ExecutorService taskExecutor = new ThreadPoolBuilder().withUnboundedQueue()
+                .withCorePoolSize(ENUM_VALIDATOR_THREADS)
+                .withMaxPoolSize(ENUM_VALIDATOR_THREADS)
+                .withName("Enum Validator Task Executor")
+                .build();
+
+        for (final Locator locator : locators) {
+            // start async read for enum values of locator from cassandra
+            Future<ColumnList<Long>> enumHashValuesFuture = taskExecutor.submit(new Callable() {
+                @Override
+                public ColumnList<Long> call() throws Exception {
+                    return AstyanaxReader.getInstance().getColumnsFromEnumCF(locator);
+                }
+
+            });
+
+            // start async read for enum values of locator from elastic search
+            Future<List<SearchResult>> esEnumValuesFuture = taskExecutor.submit(new Callable() {
```

Comment:
We decided to just make sync read for both the cassandra and ES retrievals, eliminating the taskExecutor threadpool entirely here.  And only read from ES after we verified that the metric is good, and use the results to check whether to update it with the cassandra definitive values.